app/models/spree/variant.rb

```diff
@@ -29,9 +29,9 @@ class Variant < ApplicationRecord
 
     belongs_to :product, -> { with_deleted }, touch: true, class_name: 'Spree::Product'
     belongs_to :tax_category, class_name: 'Spree::TaxCategory'
+    belongs_to :shipping_category, class_name: 'Spree::ShippingCategory'
```

Comment:
```suggestion
    belongs_to :shipping_category, class_name: 'Spree::ShippingCategory', optional: false
```

Instead of the presence validation.